src/engine/prague.md

```diff
@@ -35,7 +35,7 @@ Method parameter list is extended with `executionRequests`.
   1. `executionPayload`: [`ExecutionPayloadV3`](./cancun.md#executionpayloadv3).
   2. `expectedBlobVersionedHashes`: `Array of DATA`, 32 Bytes - Array of expected blob versioned hashes to validate.
   3. `parentBeaconBlockRoot`: `DATA`, 32 Bytes - Root of the parent beacon block.
-  4. `executionRequests`: `Array of DATA` - List of execution layer triggered requests. Each list element is the corresponding request type's `request_data` as defined by [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685). Elements of the list **MUST** be ordered by `request_type` in ascending order.
+  4. `executionRequests`: `Array of DATA` - List of execution layer triggered requests. Each list element is the corresponding request type's `requests` as defined by [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685). Elements of the list **MUST** be ordered by `request_type` in ascending order. Elements with empty `request_data` are excluded from the list.
```

Comment:
If we remove empty elements from the array, how are we gonna the elements back to each request type? Currently, the assumption is that the element position is the identifier of their type.

Reply:
For reference, Teku's implementation of the decoding:
https://github.com/Consensys/teku/blob/master/ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/versions/electra/ExecutionRequestsDataCodec.java#L40-L78

Reply:
Yes, great comment, and also for completeness I would go back to `request_type ++ request_data`, because that re-opens the possibility to have multiple `request_data` of the same type.

Reply:
What is the use case of multiple `request_data` of the same type in the engine API requests list?

Reply:
My intention here is going back to `request_type` ++ `request_data` for each element. This is why I changed it back to being a list of `requests` instead of a list of `request_data` items.

The type byte is needed to distinguish the items when there are missing ones. I would not go as far as allowing multiple list items for a `request_type`. In fact it doesn't work for the CL because it needs to be able to map back.

Reply:
The rules on the CL are the same as we had in previous iterations. One element is allowed per type byte, and the types have to be given in increasing order. In practice this means you simply need to check that the `request_type` of the element is `>` the one of the previous element. That's it.

Reply:
> This is why I changed it back to being a list of requests instead of a list of request_data items

You are correct. When I read it the first time I didn't notice the reference to `request` instead of `request_data`.

I think making the change to include the type and remove empty elements is fine. The other suggested change to allow many elements of the same type I think complicates the logic for no concrete benefit.

Reply:
This change feels like byte shedding.